### PR TITLE
update TOC to match rocDecode

### PR DESCRIPTION
**Pull request by randyh62**
resolves https://ontrack-internal.amd.com/browse/ROCDOC-416

Summary of proposed changes:
- updated sphinx/_toc.yml.in to match index.rst
- updated index.rst to match TOC

### CODE REVIEW

File: docs/sphinx/_toc.yml.in [hunk 1]
```diff
@@ -2,13 +2,18 @@ defaults:
   numbered: False
 root: index
 subtrees:
-- entries: 
+- caption: Installation
```
Comment (Rmalavally) [inline docs/sphinx/_toc.yml.in:5]: Why did we change this to caption, Randy?

Comment (randyh62) [reply docs/sphinx/_toc.yml.in:5]: Just following the rocDecode structure as recommended.

**Review by Rmalavally — APPROVED** [review]
Looks good, Randy.

File: docs/index.rst [hunk 2]
```diff
@@ -21,13 +21,13 @@ The hipBLAS documentation is structured as follows:
 
     * :ref:`install`
 
-  .. grid-item-card:: How-To
+  .. grid-item-card:: How to
```
Comment (daineAMD) [inline docs/index.rst:24]: I liked it better hyphenated, but I'm not an English expert or anything so I could just be wrong haha

**Review by daineAMD — APPROVED** [review]

File: docs/index.rst (hunk 2 above)
Comment (randyh62) [reply docs/index.rst:24]: Mine is not to question why...

**Pull request merged**